_posts/2019-04-25-snapshots.md

```diff
@@ -0,0 +1,135 @@
+---
+layout: post
+title: "Take <i>snapshots</i> of current screen"
+author: "beibean"
+cover: /assets/snapshots/cover.jpg
+image:
+  path: /assets/snapshots/cover.jpg
+  height: 300
+  width: 300
+---
+
+## What is this post about
+
+This post rises to explain how to take a snapshot of what LittlevGL is printing in the whole screen. In other words, press a pseudo *Ptr Scr* key.
+
+I share my way of doing it (with holder functions/variables). I am sure that this way will not fit anyone code at a first glance, but I hope that you catch the drift, so you are able to adapt it for your needs.
+
+## Why would I want to take a screen snapshot
+
+This is basically for two main reasons:
+
+- It helps to analyze what is currently being shown within the screen in a pixel level. For example, opening the image with GIMP could help us to realize that we should move elements on GUI, change colors, modify themes...
+
+- It is a must if you want to have automatic tests in your program. This could be achieved simply by comparing a reference and new output (it will be explained later).
+
+## How to obtain a binary of the whole VDB
+
+The first thing we want to achieve is to flush to a binary file the screen pixel values:
+
+- Global variables used:
+
+```c
+static FILE* pFileOut;
+static BOOL_E bSnapshot;
+```
+
+- Create a function which will be called whenever we want to take a snapshot:
+
+```c
+/** Flush buffer.
+ *
+ * Flush buffer into a binary file.
+ *
+ * @note: data pixel should be formated to uint32_t RGBA. Imagemagick requirements.
+ *
+ * @param[in] pFileName   Output binary file name.
+ *
+ */
+void DISP_IMPL_flushBuffer (const char_t* pFileName)
+{
+  pFileOut = fopen(pFileName, "w");
+
+  if (pFileOut == NULL) {
+    SIM_logMsg(("[Display] error: %s cannot be opened", pFileName));
```

Comment:
Please use one of the `lv_log` functions instead, rather than introducing a non-standard function. `printf` would also be acceptable since `fopen` is being used.